### ISSUE TITLE
Go to definition does not work with Jinja Template files

### ISSUE THREAD

**Issue by DonJayamanne** [microsoft/python-language-server#1136]
@rbhanot4739 commented on [Mon May 27 2019](https://github.com/microsoft/vscode-python/issues/5795)

## Environment data

- VS Code version: 1.34.0
- Extension version (available under the Extensions sidebar): 2019.4.2 
- OS and version: Ubuntu 16.04 LTS
- Python version (& distribution if applicable, e.g. Anaconda): Python 2.7 and Python 3.6
- Type of virtual environment used (N/A | venv | virtualenv | conda | ...): virtualenv 
- Relevant/affected Python packages and their versions: XXX

I am working on Python Flask/Django project which has number of Jinja Template files. The native Go to Definition/Declaration functionality does not work with Jinja files. I am not sure if it is a problem with Python extension not supporting Jinja files or something but I created the issue for both VsCode and Jinja Plugin and I was advised to report the issue on official python extension.

## Expected behaviour

Go to definition/Peek definition should work for symbols in html jinja files.

## Actual behaviour

I get a popup balloon saying no definition could be found.

## Steps to reproduce:
I have the following snippet from my Jinja template file.
```
{% extends "base.html" %}
{% from "_posts_jinja.html" import delete_post %}
{% block body %}
```
Now If I try to jump to let's say base.html or _posts_jinja.html file or to delete_post macro which has been defined inside the _posts_jinja.html, VsCode shows a popup balloon saying "no definition can be found" for this symbol.

Infact these symbols in Jinja/html files are not even recognized or indexed because if I hit ctrl-t and try to look up for delete_post macro, it does not show up. However the other symbols in the python files show up just fine.

Let me know if some more information is required from myside.

**Comment by MikhailArkhipov** [comment]
LS does not handle HTML content. I believe HTML handling comes standard in VS Code. Not sure if it supports navigation though.

**Comment by DonJayamanne** [comment]
> Python extension does not register for HTML.

Agreed. However code navigation for Python code (jinja templates) is something that would be considered an enhancement.
Though there are other challenges as well, such as:
* Does this mean, whenever a `html` file is opened, the python extension gets activated?

On the other hand, if the extension of the jinja template is `jinja2` or similar, then it might be acceptable for the python extension to get activated.

**Comment by jakebailey** [comment]
How does VSC's HTML editor handle this? I know you can embed JS anywhere it's allowed and get completion (in `<script>`, even `onClick`/`onSubmit` tags).

**Comment by MikhailArkhipov** [comment]
They have combined parser that identifies language blocks, then in VSC they internally redirect to multiple language services (most of which they also own) or forward to external LS like with C#. This is basically manual implementation of Visual Studio core editor [projections](https://docs.microsoft.com/en-us/visualstudio/extensibility/inside-the-editor?view=vs-2019#projection).
